libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -1024,8 +1027,9 @@ void AP_Mount::convert_params()
             // conversion is still done even if HAL_MOUNT_SERVO_ENABLED is false
             mnt_type = 7;  // (int8_t)Type::BrushlessPWM;
         }
+        // if the mount was previously set, then we need to save the upgraded mount type
+        _params[0].type.set_and_save(mnt_type);
     }
-    _params[0].type.set_and_save(mnt_type);
```

Comment:
Moving this inside the else ensures that we only write a value to `MNT1_TYPE` if the previous `MNT_TYPE` was set to a non zero value.

ie. don't set `MNT1_TYPE` to a configured value of zero, as this is no longer "default".